### PR TITLE
feat(managers): make StorageManager query peers from storage

### LINKED ISSUE #98: Recover peer addresses from local storage
#### Context
- As explained in #97, peer addresses obtained from our peers through the peer discovery protocol need to be persisted from volatile memory into local storage to enhance trustlessness of future cold-bootstrapping when restarting the node.
- Loading peer addresses from storage needs only to happen upon node startup. That is, it's a fundamental part of the "heat up" routine of the `PeersManager` actor.

#### Actionables
- Make `PeersManager` query `StorageManager` for the peer addresses list upon startup and put its content into its own state by means of calling itself with `AddPeers`, so that the pertinent validations can be run on the peer addresses.

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Closes #98

### CODE REVIEW

**Comment by tmpolaczyk** [comment]
Blocked on #113

File: config/src/lib.rs [hunk 1]
```diff
@@ -110,6 +112,10 @@ impl ConnectionsConfig {
     pub fn default_known_peers() -> Vec<SocketAddr> {
         Vec::default()
     }
+
+    fn default_storage_peers_period() -> u64 {
+        30
```
Comment (aesedepece) [inline config/src/lib.rs]: This is a magic number, please:
- Move it to a more visible spot
- Make it static
- Add a comment for it, potentially reasoning about considerations when deciding the value

File: core/src/actors/peers_manager.rs [hunk 2]
```diff
@@ -20,6 +25,42 @@ pub struct PeersManager {
     peers: Peers,
 }
 
+impl PeersManager {
+    /// Method to periodically dump peers into storage
+    fn storage_peers(&self, ctx: &mut Context<Self>, storage_peers_period: u64) {
+        // Schedule the discovery_peers with a given period
+        ctx.run_later(
+            Duration::from_secs(storage_peers_period),
+            move |act, ctx| {
```
Comment (aesedepece) [inline core/src/actors/peers_manager.rs]:
```suggestion
            move |actor, ctx| {
```

File: core/src/actors/peers_manager.rs [hunk 3]
```diff
@@ -20,6 +25,42 @@ pub struct PeersManager {
...
+                // Get storage manager address
+                let storage_manager_addr = System::current().registry().get::<StorageManager>();
+
+                // Dump peers into storage. `AsyncContext::wait` register
+                // future within context, but context waits until this future resolves
+                // before processing any other events.
+                storage_manager_addr
+                    .send(Put::from_value(PEERS_KEY, &act.peers).unwrap())
+                    .into_actor(act)
+                    .then(|res, _act, ctx| {
```
Comment (aesedepece) [inline core/src/actors/peers_manager.rs]:
```suggestion
                    .then(|res, _actor, ctx| {
```

File: core/src/actors/peers_manager.rs [hunk 4]
```diff
@@ -29,15 +70,69 @@ impl Actor for PeersManager {
         debug!("Peers Manager actor has been started!");
 
         // Send message to config manager and process response
-        send_get_config_request(self, ctx, |s, _ctx, config| {
+        send_get_config_request(self, ctx, |act, ctx, config| {
```
Comment (aesedepece) [inline core/src/actors/peers_manager.rs]:
```suggestion
        send_get_config_request(self, ctx, |actor, ctx, config| {
```

File: core/src/actors/storage_manager.rs [hunk 5]
```diff
@@ -13,6 +13,9 @@ use witnet_util::error::WitnetError;
 type ValueStorageResult<T> = StorageResult<Option<T>>;
 type UnitStorageResult = StorageResult<()>;
 
+/// Constant to specify the peers key for the storage
+pub static PEERS_KEY: &'static [u8] = b"peers";
```
Comment (aesedepece) [inline core/src/actors/storage_manager.rs]: I don't think this constant belongs here. We may need a separate file somewhere for static storage keys.

File: docs/architecture/managers/peers-manager.md [hunk 6]
```diff
@@ -92,21 +92,36 @@ The way other actors will communicate with the storage manager is:
...
 These are the messages sent by the peers manager:
 
-| Message   | Destination   | Input type | Output type                 | Description               |
-| --------- | ------------- | ---------- | --------------------------- | ------------------------- |
-| GetConfig | ConfigManager | `()`       | `Result<Config, io::Error>` | Request the configuration |
-
-It is foreseen that in the future, the peers manager will call the `Storage Manager` to store a list of known peers.
+| Message   | Destination     | Input type                                | Output type                 | Description                               |
+| --------- | --------------- | ----------------------------------------- | --------------------------- | ----------------------------------------- |
+| GetConfig | ConfigManager   | `()`                                      | `Result<Config, io::Error>` | Request the configuration                 |
```
Comment (aesedepece) [inline docs/architecture/managers/peers-manager.md]:
```suggestion
| `GetConfig` | `ConfigManager`   | `()`                                      | `Result<Config, io::Error>` | Request the configuration                 |
```

File: docs/architecture/managers/peers-manager.md [hunk 7]
```diff
@@ -92,21 +92,36 @@ The way other actors will communicate with the storage manager is:
...
+| Get       | StorageManager  | `&'static [u8]`                           | `StorageResult<Option<T>>`  | Wrapper to RocksStorage `get()` method    |
```
Comment (aesedepece) [inline docs/architecture/managers/peers-manager.md]:
```suggestion
| `Get`       | `StorageManager`  | `&'static [u8]`                           | `StorageResult<Option<T>>`  | Wrapper to RocksStorage `get()` method    |
```

File: docs/architecture/managers/peers-manager.md [hunk 8]
```diff
@@ -92,21 +92,36 @@ The way other actors will communicate with the storage manager is:
...
+| Put       | StorageManager  | `&'static [u8]`, `Vec<u8>`                | `StorageResult<()>`         | Wrapper to RocksStorage `put()` method    |
```
Comment (aesedepece) [inline docs/architecture/managers/peers-manager.md]:
```suggestion
| `Put`       | `StorageManager`  | `&'static [u8]`, `Vec<u8>`                | `StorageResult<()>`         | Wrapper to RocksStorage `put()` method    |
```

File: docs/architecture/managers/peers-manager.md (hunk 7 above)
Comment (aesedepece) [inline docs/architecture/managers/peers-manager.md]: I don't feel comfortable with the mention to `RocksStorage` here. This actor shouldn't give a ʞɔnɟ for the concrete storage backend we use, and neither should its documentation.

File: config/src/lib.rs (hunk 1 above)
Comment (csmoe) [reply config/src/lib.rs]: and is it appropriate to mark this method `#[inline}`?

Comment (aesedepece) [reply config/src/lib.rs]: Sounds good! :+1:

File: docs/architecture/managers/peers-manager.md (hunk 7 above)
Comment (tmpolaczyk) [reply docs/architecture/managers/peers-manager.md]: :+1: Replaced `RocksStorage` with `Storage`

**Pull request merged**